File: Easy/Geek Jump/geek-jump.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG{
    public static void main(String args[]) throws IOException{
        Scanner sc=new Scanner(System.in);
        int t=sc.nextInt();
        while(t-->0){
            int N=sc.nextInt();
            int []arr=new int[N];
            for(int i=0;i<N;i++){
                arr[i]=sc.nextInt();
                
            }
            Solution obj=new Solution();
            int res=obj.minimumEnergy(arr,N);
            System.out.println(res);
            
        }
    }
}


// } Driver Code Ends


//User function Template for Java

class Solution{
    public int minimumEnergy(int h[],int n){
        int[] arr= new int[n];
        arr[0]=0;
        if(n>1)
        arr[1]=Math.abs(h[1]-h[0]);
        for(int i=2;i<n;i++)
        {
            arr[i]=Math.min(arr[i-1]+Math.abs(h[i]-h[i-1]),
                            arr[i-2]+Math.abs(h[i]-h[i-2]));
        }
        return arr[n-1];
    }
}

File: Easy/Geek Jump/geek-jump.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    int minimumEnergy(vector<int>& height, int n) {
        vector<int> arr(n,-1);
        arr[0]=0;
        arr[1]=abs(height[0]-height[1]);
        for(int i=2;i<n;i++)
        {
            arr[i]=min(arr[i-1]+abs(height[i]-height[i-1]),arr[i-2]+abs(height[i]-height[i-2]));
        }
        return arr[n-1];
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        cin >> N;
        vector<int> arr(N);
        for (int i = 0; i < N; i++) {
            cin >> arr[i];
        }
        Solution obj;
        cout << obj.minimumEnergy(arr, N) << "\n";
    }
    return 0;
}
// } Driver Code Ends